src/provider/model/AppDirectory.ts

```diff
@@ -35,9 +35,12 @@ export class AppDirectory extends AsyncInit {
         }) || null;
     }
 
-    public async getAppsByIntent(intentType: string): Promise<Application[]> {
+    public async getAppsByIntent(intentType: string, contextType?: string): Promise<Application[]> {
```

Comment:
I think I might be missing some info on that one. I thought where there's an empty list of context types it should mean the intent is interested in all contexts?

Reply:
Could well be the case, any idea where that came from? Though it's not consistent with the behaviour of `getAppIntentsByContext`.

Reply:
It's purely a guess actually. I couldn't think of any logical reasons why contexts being an empty array and contexts being undefined should be different in behaviour?

Reply:
Think we might be at crossed wires here. Let's discuss in person

Reply:
Just to summarise earlier conversation - we should treat an empty array in the same way as an `undefined` contexts list. This is because otherwise the intent definition would be unusable, and we would have `findIntents` (when called without a context type) returning intent definitions that aren't possible to actually use with `raiseIntent`.

Reply:
👍